addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -1989,20 +1996,30 @@ export class OdooEditor extends EventTarget {
         const isRemovableInvisible = (node, noBlocks = true) =>
             !isVisible(node, noBlocks) && !isUnremovable(node);
         const endIsStart = end === start;
-        while (end && isRemovableInvisible(end, false) && !end.contains(range.endContainer)) {
+        while (end && isRemovableInvisible(end, false) && !end.contains(range.endContainer) && !shouldMergeLi) {
             const parent = end.parentNode;
             end.remove();
             end = parent;
         }
+        if (shouldMergeLi) {
+            fillEmpty(closestElement(end, 'li'));
+        }
         // Same with the start container
+
         while (
             start &&
-            isRemovableInvisible(start) &&
+            (isRemovableInvisible(start) || shouldMergeLi) &&
             !(endIsStart && start.contains(range.startContainer))
         ) {
             const parent = start.parentNode;
             start.remove();
             start = parent;
+            if (shouldMergeLi &&
+                !((start.nodeName === 'LI' || start.nodeName === 'OL') &&
+                    (isZWS(start) || !start.hasChildNodes()))
+            ) {
```

Comment:
Which can be written as
```suggestion
            if (shouldMergeLi && (!['UL', 'OL', 'LI'].includes(start.nodeName) || (start.hasChildNodes() && !isZWS(start)))) {
```